File: app/src/main/java/com/lambdaschool/sprint4challenge_mymovies/MovieAdapter.java
package com.lambdaschool.sprint4challenge_mymovies;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.lambdaschool.sprint4challenge_mymovies.apiaccess.MovieOverview;

import java.util.ArrayList;

class MovieAdapter extends RecyclerView.Adapter<MovieAdapter.ViewHolder> {

    ArrayList<MovieOverview> movieList;
    public Context context;

    public MovieAdapter(ArrayList<MovieOverview> movies, Context context) {
        this.movieList = movies;
        this.context = context;
    }


    public MovieAdapter() {
        super();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.movie_layout, viewGroup,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int i) {

        //        String name = movieList.get(i);                                               // not sure of this part, might need to check red lightbulb for suggestions
        String name = String.valueOf(movieList.get(i));
        viewHolder.nameView.setText(name);

    }


    @Override
    public int getItemCount() {
        return movieList.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder{


        TextView nameView;

        LinearLayout linearLayout;


        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            linearLayout = itemView.findViewById(R.id.linear_layout);
            nameView = itemView.findViewById(R.id.name_view);

        }
    }

}

File: app/src/main/java/com/lambdaschool/datapersistencesprintchallenge/recycler/MovieAdapter.kt
package com.lambdaschool.datapersistencesprintchallenge.recycler

import android.graphics.Movie
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.lambdaschool.datapersistencesprintchallenge.R
import com.lambdaschool.sprint4challenge_mymovies.model.MovieOverview
import kotlinx.android.synthetic.main.movie_list_item.view.*

class MovieAdapter (val data: MutableList<MovieOverview>):
        RecyclerView.Adapter<MovieAdapter.ViewHolder>(){
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val viewGroup = LayoutInflater.from(parent.context).inflate(R.layout.movie_list_item, parent, false)

            return ViewHolder(viewGroup)
    }

    override fun getItemCount(): Int {
        return data.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
      val movieItems = data[position]
        holder.bindModel(movieItems)

        holder.listParent.setOnClickListener {
            if (movieItems.isVaf){
                movieItems.isVaf = false
                notifyItemChanged(position)
            }else{
                movieItems.isVaf = true
                notifyItemChanged(position)
            }
        }


    }

    class ViewHolder(view: View):RecyclerView.ViewHolder(view){

        var id = view.movie_id_text
        var title = view.movie_title_text
        val listParent = view.movie_list
        var lastPosition = -1

        fun bindModel(list: MovieOverview){
            id.text = list.id.toString()
            title.text = list.title

            if (list.isVaf){
                listParent.setBackgroundColor(ContextCompat.getColor(itemView.context, R.color.colorAccent))
            }else
                listParent.setBackgroundColor(ContextCompat.getColor(itemView.context, R.color.white))

        }





    }
}